File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/maximum-number-of-operations-to-move-ones-to-the-end.go
func maxOperations(binaryString string) int {
	countOnes := int(binaryString[0] - '0')
	maxOperations := 0

	for i := 1; i < len(binaryString); i++ {
		countOnes += int(binaryString[i] - '0')
		maxOperations += Ternary(binaryString[i-1] > binaryString[i], countOnes, 0)
	}

	return maxOperations
}

func Ternary[T any](condition bool, first T, second T) T {
	if condition {
		return first
	}

	return second
}

File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/maximum-number-of-operations-to-move-ones-to-the-end.java
public class Solution {
    public int maxOperations(String binaryString) {
        char[] binaryArray = binaryString.toCharArray();
        int countOnes = binaryArray[0] - '0';
        int maxOperations = 0;

        for (int i = 1; i < binaryArray.length; ++i) {
            countOnes += binaryArray[i] - '0';
            maxOperations += binaryArray[i - 1] > binaryArray[i] ? countOnes : 0;
        }
        
        return maxOperations;
    }
}